### ISSUE TITLE
Conv2D produces inconsistent results between GPU and CPU (large relative error ~9.59e+02)

### ISSUE THREAD

**Issue by zifan6699** [tensorflow/tensorflow#117677]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

binary

### TensorFlow version

2.22.0-dev20260302

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04 (Docker environment)

### Mobile device

N/A

### Python version

3.10

### Bazel version

N/A

### GCC/compiler version

N/A

### CUDA/cuDNN version

12.5.1/9.2.1

### GPU model and memory

NVIDIA GeForce RTX 3090, 24GB

### Current behavior?

Current behavior:

Running the same Conv2D layer (filters=64, kernel_size=7, strides=2, padding="valid", 
use_bias=False) with identical float32 weights and identical float32 input on GPU vs CPU 
produces significantly different numerical results:
  MaxAbsErr : 1.189858e-04
  MaxRelErr : 9.591902e+02
This far exceeds the expected FP32 precision bound of 2^-23 ≈ 1.19e-7.
No NaN or Inf values are present in either output — the discrepancy is purely numerical.
Expected behavior: GPU and CPU should produce results within FP32 rounding tolerance 
(relative error ≤ 2^-23). A relative error of ~9.59e+02 indicates fundamentally different 
computation paths or non-deterministic kernel selection between devices.
Here are the external files required for reproducing this issue：
[test1.zip](https://github.com/user-attachments/files/27405522/test1.zip)

### Standalone code to reproduce the issue

```shell
import os, numpy as np, tensorflow as tf

HERE    = os.path.dirname(os.path.abspath(__file__))
inp     = np.load(os.path.join(HERE, "input.npz"))["data"]
kernel  = np.load(os.path.join(HERE, "weights.npz"))["kernel"]

# Conv2D(64, kernel_size=7, strides=2, padding="valid", use_bias=False)
# Matches DenseNet stem conv0 exactly.

THR = 2.0 ** -23

print(f"Reproduction  (Input: {inp.shape}, batch={inp.shape[0]})")

def run(device):
    tf.keras.backend.clear_session()
    with tf.device(f"/{device.upper()}:0"):
        layer = tf.keras.layers.Conv2D(64, 7, strides=2, padding="valid", use_bias=False)
        layer(tf.zeros((1,) + inp.shape[1:], dtype=tf.float32))
        layer.set_weights([kernel])
        return layer(tf.constant(inp), training=False).numpy()

out_gpu = run("gpu")
out_cpu = run("cpu")

abs_err = np.max(np.abs(out_gpu - out_cpu))
rel_err = np.max(np.abs(out_gpu - out_cpu) / (np.abs(out_gpu) + 1e-12))
fail    = rel_err > THR or np.isnan(rel_err)

print(f"MaxAbsErr : {abs_err:.6e}")
print(f"MaxRelErr : {rel_err:.6e}")
print(f"Status    : {'FAIL' if fail else 'PASS'}")
```

### Relevant log output

```shell
Reproduction  (Input: (16, 230, 230, 3), batch=16)
I0000 00:00:1777992567.599561   18662 gpu_device.cc:2042] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 20007 MB memory:  -> device: 0, name: NVIDIA GeForce RTX 3090, pci bus id: 0000:01:00.0, compute capability: 8.6
I0000 00:00:1777992567.610348   18662 gpu_device.cc:2042] Created device /job:localhost/replica:0/task:0/device:GPU:1 with 18305 MB memory:  -> device: 1, name: NVIDIA GeForce RTX 3090, pci bus id: 0000:02:00.0, compute capability: 8.6
I0000 00:00:1777992570.196881   18662 cuda_dnn.cc:461] Loaded cuDNN version 92101
MaxAbsErr : 1.189858e-04
MaxRelErr : 9.591902e+02
Status    : FAIL
```
